lib/router/with-router.js

```diff
@@ -0,0 +1,32 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import hoistStatics from 'hoist-non-react-statics'
+
+export default function withRouter (options = {}) {
+  const { routerPropName = 'router' } = options
+
+  return function withRouterComponent (ComposedComponent) {
+    class WithRouterWrapper extends Component {
+      static contextTypes = {
+        router: PropTypes.object
+      }
+
+      static displayName = `withRouter(${ComposedComponent.displayName})`
```

Comment:
@chaffeqa Added! Out of curiosity, what's the difference between displayName and name? I see [displayName](https://facebook.github.io/react/docs/react-component.html#displayname) in the React docs but not name.